src/main/NWNX/API/Events/PartyEvents.cs

```diff
@@ -0,0 +1,329 @@
+using NWN.API;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events
+{
+  public class PartyEvents
+  {
+    [NWNXEvent("NWNX_ON_PARTY_LEAVE_BEFORE")]
+    public class OnPartyLeaveBefore : EventSkippable<OnPartyLeaveBefore>
+    {
+      /// <summary>
+      /// Gets the player doing the action.
+      /// </summary>
+      public NwPlayer Player { get; private set; }
+
+      /// <summary>
+      /// Gets NwCreature promoted to party leader.
+      /// </summary>
+      public NwCreature Leaving { get; private set; }
```

Comment:
I have not tested this, but is `Player` and `Leaving` the same object here? Is it always a player, or is this fired for summons/henchmen creatures too?
```suggestion
      /// <summary>
      /// Gets the player leaving the party.
      /// </summary>
      public NwPlayer Player { get; private set; }
```